Example/Custom Integration (ObjC)/ThreeDSPaymentIntentExampleViewController.m

```diff
@@ -0,0 +1,133 @@
+ PaymentIntents.
+
+ 1. Collect user's card information via `STPPaymentCardTextField`
+ 2. Create a `PaymentIntent` on our backend (this can happen concurrently with #1)
+ 3. Confirm PaymentIntent using the `STPSourceParams` for the user's card information.
+ 4. If the user needs to go through the 3D Secure authentication flow, use `STPRedirectContext` to do so.
+ 5. When user returns to the app, or finishes the SafariVC redirect flow, `STPRedirectContext` notifies via callback
+
+ See the documentation at https://stripe.com/docs/payments/dynamic-authentication for more information
+ on using PaymentIntents for dynamic authentication.
+ */
+@interface ThreeDSPaymentIntentExampleViewController ()
+
+@end
+
+@implementation ThreeDSPaymentIntentExampleViewController
+
+- (void)pay {
+    if (![self.paymentTextField isValid]) {
+        return;
+    }
+    if (![Stripe defaultPublishableKey]) {
+        [self.delegate exampleViewController:self didFinishWithMessage:@"Please set a Stripe Publishable Key in Constants.m"];
+        return;
+    }
+    [self updateUIForPaymentInProgress:YES];
+
+    NSString *returnUrl = @"payments-example://stripe-redirect";
+    // In a more interesting app, you'll probably create your PaymentIntent as soon as you know the
+    // payment amount you wish to collect from your customer. For simplicity, this example does it once they've
+    // pushed the Pay button.
+    // https://stripe.com/docs/payments/dynamic-authentication#create-payment-intent
+    [self.delegate createBackendPaymentIntentWithAmount:@1099 returnUrl:returnUrl completion:^(STPBackendResult status, NSString *clientSecret, NSError *error) {
+        if (status == STPBackendResultFailure || clientSecret == nil) {
+            [self.delegate exampleViewController:self didFinishWithError:error];
+            return;
+        }
+
+        STPAPIClient *stripeClient = [STPAPIClient sharedClient];
+        STPPaymentIntentParams *paymentIntentParams = [[STPPaymentIntentParams alloc] initWithClientSecret:clientSecret];
+        paymentIntentParams.sourceParams = [STPSourceParams cardParamsWithCard:self.paymentTextField.cardParams];
+
+        [stripeClient confirmPaymentIntentWithParams:paymentIntentParams completion:^(STPPaymentIntent * _Nullable paymentIntent, NSError * _Nullable error) {
+            if (error) {
+                [self.delegate exampleViewController:self didFinishWithError:error];
+                return;
+            }
+
+            if (paymentIntent.status == STPPaymentIntentStatusRequiresSourceAction) {
+                self.redirectContext = [[STPRedirectContext alloc] initWithPaymentIntent:paymentIntent returnUrl:returnUrl completion:^(NSString * _Nonnull clientSecret, NSError * _Nullable error) {
+                    if (error) {
+                        [self.delegate exampleViewController:self didFinishWithError:error];
+                    }
+                    else {
```

Comment:
this file is mixing `} else {` and `}\n else {` Do we have a standard in our sample code? I'd vote for `} else {`

Reply:
We encoded `}\n else {` in our style guide a long time ago but I'm coming around to doing `} else {` everywhere. Should open a PR that fixes this across the codebase.

Reply:
https://github.com/stripe/stripe-ios/blob/master/STYLEGUIDE.md#control-flow

Reply:
I don't think I've ever seen `}\n else {` in a style guide, and it wouldn't be my choice, but I try to adhere to it. Thanks for catching the mistake.